### PR TITLE
feat: add <Tray> for responsiveness in <Menu> and <Select>

### PR DESCRIPTION
**Pull request by sarahgm**

### CODE REVIEW

File: packages/components/src/theme.ts [hunk 1]
```diff
@@ -35,6 +35,7 @@ import type { SwitchThemeExtension } from './Switch';
 import type { TableThemeExtension } from './Table';
 import type { TextThemeExtension } from './Text';
 import type { TextAreaThemeExtension } from './TextArea';
+import type { TrayThemeExtension } from './Overlay';
```
Comment (sebald) [inline packages/components/src/theme.ts:38]: Could add this to the overlay above :)

File: packages/components/src/Menu/MenuTrigger.tsx [hunk 2]
```diff
@@ -5,6 +5,8 @@ import { MenuContext, MenuContextProps } from './Context';
 import { PressResponder } from '@react-aria/interactions';
 import { Popover } from '../Overlay/Popover';
 import { useObjectRef } from '@react-aria/utils';
+import { useResponsiveValue } from '@marigold/system';
+import { Tray } from '../Overlay';
```
Comment (sebald) [inline packages/components/src/Menu/MenuTrigger.tsx:9]: import with Popover? :)

File: themes/theme-b2b/src/components/Tray.style.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Theme } from '@marigold/components';
+
+export const Tray: Theme['components']['Tray'] = {
+  base: {
```
Comment (sebald) [inline themes/theme-b2b/src/components/Tray.style.ts:4]: If we have to add this to every theme, maybe we should add this as `__baseCSS` to the `Tray`?

Comment (sarahgm) [reply themes/theme-b2b/src/components/Tray.style.ts:4]: yes, so should we keep the possibility to style or remove the themeextension possibility?

**Review by sebald — APPROVED** [review]

**Pull request merged**